lib/qiita/events/main.ex

```diff
@@ -0,0 +1,9 @@
+defmodule Qiita.Events.Main do
```

Comment:
I think `Main` can be more descriptive name. What about something like `MarkdownGenerator`? Or maybe we could move the main run function to the beheviour module. That should be fine. [GenServer](https://github.com/elixir-lang/elixir/blob/89e57882e120e77d755477b96ecfa15032dd582d/lib/elixir/lib/gen_server.ex) module defines some functions as well as behavour callbacks.

```elixir
defmodule Qiita.Events.Markdown do
  @type items :: list()

  @callback generate(items) :: String.t()
  
  import Qiita.Events.Repo

  def run(query, start_time, end_time, mod, tags, title, id) do
    fetch_items(query, start_time, end_time)
    |> mod.generate()
    |> post_markdown(tags, title, id)
  end
end
```